soccer/src/soccer/strategy/agent/position/runner.cpp

```diff
@@ -0,0 +1,72 @@
+#include "runner.hpp"
+
+namespace strategy {
+
+Runner::Runner(int r_id) : Position(r_id) { position_name_ = "Runner"; }
+
+std::optional<RobotIntent> Runner::derived_get_task(RobotIntent intent) {
+    current_state_ = update_state();
+    return state_to_task(intent);
+}
+
+Runner::State Runner::update_state() {
+    State next_state = current_state_;
+    // handle transitions between states
+    WorldState* world_state = this->world_state();
+    rj_geometry::Point robot_position = world_state->get_robot(true, robot_id_).pose.position();
+
+    switch (current_state_) {
+        case RIGHT:
+            if (robot_position.x() <= field_dimensions_.field_x_left_coord() + 1.1) next_state = UP;
```

Comment:
note the GH actions warnings here. inline if statements are hard to read! it's fine for now, but won't be approved in the future